Source/DafnyCore/Verifier/Translator.ClassMembers.cs

```diff
@@ -1353,6 +1355,49 @@ public partial class Translator {
       builder.Add(Assert(tok, q, new PODesc.TraitFrame(m.WhatKind, true), kv));
     }
 
+    // Return a way to know if an assertion should be converted to an assumptions
+    private void SetAssertionOnlyFilter(Node m) {
+      List<RangeToken> rangesOnly = new List<RangeToken>();
+      m.Visit(node => {
+        if (node is AssertStmt assertStmt && Attributes.Find(assertStmt.Attributes, "only") is UserSuppliedAttributes attribute) {
+          var (isBefore, isAfter) = attribute.Args.Count == 1 && attribute.Args[0] is LiteralExpr { Value: var value } &&
+                                  value is "after" or "before" ? (value is "before", value is "after") : (false, false);
```

Comment:
If it were me, I'd move the logic for extracting the `{:only}` attribute to a separate definition. Then maybe you'd have a datatype `AssertOnlyKind = After | Before | Only` and a `bool HasAssertOnlyAttribute(out var assertOnlyKind)`, or something like that.